src/plugins/github/nodes.js

```diff
@@ -125,6 +127,12 @@ export function fromRaw(x: RawAddress): StructuredAddress {
   function fail() {
     return new Error(`Bad address: ${NodeAddress.toString(x)}`);
   }
+  if (NodeAddress.hasPrefix(x, GitNode.Prefix.commit)) {
+    const commitAddress: GitNode.CommitAddress = (GitNode.fromRaw(
```

Comment:
You’re relying on _another_ implementation detail here, which is that
the `Prefix.commit` prefix maps via `fromRaw` into the specifically the
`Commit` subtype of structured addresses. (You may object that this
obviously holds. But the Git plugin is within its rights to add
addresses like [sourcecred, git, commit, submodule_commit, repo, hash],
introducing a new structured type `SubmoduleCommit` disjoint from other
commit types.)

This comment and the one above it can both be handled by instead
writing (untested):

```js
if (NodeAddress.hasPrefix(x, GitNode.Prefix.base)) {
  const structured: GitNode.StructuredAddress = GitNode.fromRaw((x: any));
  if (structured.type === GitNode.COMMIT_TYPE) {
    return (structured: GitNode.CommitAddress);
  } else {
    throw new Error("unexpected Git node address: " + stringify(structured));
  }
}
```

Would you mind making this change?

(Relatedly: please do test the error case of providing, say, a Git tree
address.)

Reply:
Done. Added test case as requested.